### ISSUE TITLE
Unable to open tcp connection with host 'localhost:1443': dial tcp 127.0.0.1:1443

### ISSUE THREAD

**Issue by trustieee** [microsoft/go-mssqldb#170]
**Describe the bug**

Cannot connect to named MSSQL server running locally.
```
> unable to open tcp connection with host 'localhost:1443': dial tcp 127.0.0.1:1443: connect: connection refused
```

**To Reproduce**

I've tried various connection string formats:
```go
str1 := "server=localhost,mssqlserver2022;port=61821;user id=sa;password=sa;database=FooDB"
str2 := "server=localhost,mssqlserver2022;port=1433;user id=sa;password=sa;database=FooDB"
str3 := "server=localhost/mssqlserver2022;user id=sa;password=sa;database=FooDB"
str4 := "server=localhost;user id=sa;password=sa;database=FooDB"
str5 := "server=localhost:1433;user id=sa;password=sa;database=FooDB"
str6 := "server=localhost\\mssqlserver2022:1433;user id=sa;password=sa;database=FooDB"
str7 := `server=localhost\mssqlserver2022:1433;user id=sa;password=sa;database=FooDB`

// try with sqlserver
connection, err := sql.Open("sqlserver", str)
// try with mssql
connection, err := sql.Open("mssql", str)
```

I have a C# app on the side which uses this connection string, and it succeeds:
```cs
await using var connection = new SqlConnection(
"""
persist security info=false;
server=localhost\mssqlserver2022;
initial catalog=FooDB;
user id=sa;
password=sa;
""");
```

I've tried `import _ "github.com/microsoft/go-mssqldb"` and `import _ "github.com/denisenkom/go-mssqldb"`

**Expected behavior**

It should connect

### Further technical details
SQL Server v. 16.0.1105.1
Windows 10 Home

**Comment by dlevy-msft** [comment]
Hi @trustieee - It looks like str3 should work if you reverse the "\"

Try: str3 := "server=localhost\mssqlserver2022;user id=sa;password=sa;database=FooDB"

If that does not work for you then another approach is to log into the SQL instance you wish to connect to with SSMS or ADS and look at the startup messages in the SQL error log to see which port SQL is listening on. Once you know the port you can do something like:

"server=localhost,61821;user id=sa;password=sa;database=FooDB" or "server=localhost;port=61821;userid=sa;password=sa;database=FooDB"

Please let us know if that helps.

**Comment by trustieee** [comment]
Thanks @dlevy-msft ,

It seems the issue may be related to how the app is being ran. It's being ran inside WSL.
If I copy/paste my code to my windows environment, it works.

So perhaps there needs to be additional configuration to allow the app to connect to MSSQLServer running on Windows from a go app running inside WSL?

**Comment by dlevy-msft** [comment]
@trustieee - yes it does, have you run through the steps in this doc?

 https://learn.microsoft.com/en-us/windows/wsl/networking

**Comment by trustieee** [comment]
@dlevy-msft thank you, that was the issue.

For anyone else seeing this in the future, my SQL server was using a named instance (not the default name assigned to the sql server when you create/install it), and as such it had a dynamic port assigned to it. I had to retrieve the port using the SQL Server Configuration Manager, under the TCP protocol, and then had to allow that port on my Windows host firewall. I also had to retrieve the IP address of my Windows host (`ipconfig` in a terminal), and use that as part of my connection string.

In the end, my working connection string looks like this:

```go
connectionString := server=192.168.x.x\mssqlserver2022;user id=sa;password=sa;database=PeopleDB
```
